modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -154,9 +154,11 @@ class PandasDataframe(ClassLogger):
     ----------
     partitions : np.ndarray
         A 2D NumPy array of partitions.
-    index : sequence, optional
+    index : sequence or callable, optional
         The index for the dataframe. Converted to a ``pandas.Index``.
         Is computed from partitions on demand if not specified.
+        If ``callable() -> (pandas.Index, list)`` type, then the calculation will be
```

Comment:
Can we make the second return value to be optional?

Reply:
The second return value is used as index lengths in partitions. We can return None if the function cannot determine them. I don't like the idea of returning a different number of return values, it will make it difficult to process them.

Reply:
Yes, I meant to mention in the docstring that None is a possible return value for the second rval.

Reply:
Updated